app/classifier/tasks/slider/index.jsx

```diff
@@ -37,94 +37,68 @@ SliderSummary.propTypes = {
-    this.state = { value };
-    this.handleChange = this.handleChange.bind(this);
-  }
-
-  componentDidMount() {
-    const newAnnotation = Object.assign({}, this.props.annotation, { value: this.props.task.defaultValue });
-    this.props.onChange(newAnnotation);
-  }
-
-  componentWillReceiveProps(nextProps) {
-    if (this.props.annotation.value === null) {
-      const newAnnotation = Object.assign({}, this.props.annotation, { value: this.props.task.defaultValue });
-      this.props.onChange(newAnnotation);
-    } else if (this.props.annotation.value !== this.state.value) {
-      this.setState({ value: this.props.annotation.value });
-    }
-  }
-
-  handleChange(e) {
+const SliderTask = ({ task, annotation, onChange, autoFocus }) => {
+  const handleChange = (e) => {
     let value = e.target.value;
-    if (parseFloat(value, 10) > parseFloat(this.props.task.max, 10)) {
-      value = this.props.task.max;
-    } else if (parseFloat(value, 10) < parseFloat(this.props.task.min, 10)) {
-      value = this.props.task.min;
+    if (parseFloat(value, 10) > parseFloat(task.max, 10)) {
+      value = task.max;
+    } else if (parseFloat(value, 10) < parseFloat(task.min, 10)) {
+      value = task.min;
     }
-    this.setState({ value });
-    const newAnnotation = Object.assign({}, this.props.annotation, { value });
-    this.props.onChange(newAnnotation);
-  }
+    const newAnnotation = Object.assign({}, annotation, { value });
+    onChange(newAnnotation);
+  };
 
-  render() {
-    return (
-      <div>
-        <GenericTask
-          question={this.props.task.instruction}
-          help={this.props.task.help}
-          required={this.props.task.required}
-        >
-          <div className="slider-task-container full ">
-            <div className="slider-task-range">
-              <label className="answer">
-                <div>
-                  <input
-                    className="standard-input"
-                    type="range"
-                    autoFocus={this.props.autoFocus}
-                    onChange={this.handleChange}
-                    max={this.props.task.max}
-                    min={this.props.task.min}
-                    step={this.props.task.step}
-                    value={this.state.value}
-                  />
+  return (
+    <div>
+      <GenericTask
+        question={task.instruction}
+        help={task.help}
+        required={task.required}
+      >
+        <div className="slider-task-container full ">
+          <div className="slider-task-range">
+            <label htmlFor="slider-range" className="answer">
+              <div>
+                <input
+                  className="standard-input"
+                  type="range"
+                  autoFocus={autoFocus}
+                  onChange={handleChange.bind(this)}
+                  max={task.max}
+                  min={task.min}
+                  step={task.step}
+                  value={annotation.value || task.defaultValue}
```

Comment:
Will setting the value like this throw a React warning about setting a controlled input as uncontrolled (or vice versa)?